tools/pipelines/templates/include-publish-npm-package-steps.yml

```diff
@@ -103,30 +103,49 @@ steps:
           f=$packageName
         fi
         if [[ $f != "" ]]; then
-          npm publish $f $tag ${{ parameters.publishFlags }} 2>&1 | tee publish_log
-          if grep -q "npm ERR!" "publish_log"
-          then
-              if ! grep -q "You cannot publish over the previously published versions" "publish_log"
-              then
-                let t1+=1
-              else
-                echo Package has already been published.
-                local_f="${f}_local"
-                mv "$f" "$local_f"
-                package_name=$(grep -oP 'npm notice package: \K.*' "publish_log")
-                npm pack $package_name
-                if cmp -s "$f" "$local_f"
+          for i in $( seq 1 $maximumRetryIfNetworkError )
+          do
+            echo "i :" $i
+            npm publish $f $tag ${{ parameters.publishFlags }} 2>&1 | tee publish_log
+            if grep -q "npm ERR!" "publish_log"
+            then
+                if grep -q "code ENOTFOUND" "publish_log"
                 then
-                  echo Continuing as published package matches the current one that was attempting to be released.
+                  if ! $i -eq $maximumRetryIfNetworkError
+                  then
+                    echo "Network Error Detected"
+                    continue
+                  else
+                    echo "Final Network Error"
+                    exit $t1
+                  fi
                 else
-                  echo ERROR: Published package does not match the current one attempting to be released for the same version.
-                  let t1+=1
+                  if ! grep -q "You cannot publish over the previously published versions" "publish_log"
+                  then
+                    let t1+=1
+                  else
+                    echo Package has already been published.
+                    local_f="${f}_local"
+                    mv "$f" "$local_f"
+                    package_name=$(grep -oP 'npm notice package: \K.*' "publish_log")
+                    npm pack $package_name
+                    if cmp -s "$f" "$local_f"
+                    then
+                      echo Continuing as published package matches the current one that was attempting to be released.
+                    else
+                      echo ERROR: Published package does not match the current one attempting to be released for the same version.
+                      let t1+=1
+                    fi
+                    rm "$f"
+                    mv "$local_f" "$f"
+                    exit $t1
+                  fi
                 fi
-                rm "$f"
-                mv "$local_f" "$f"
-              fi
-          fi
-          rm publish_log
+            else
+              break
+            fi
+            rm publish_log
```

Comment:
`publish_log` isn't cleaned up when "break" on line 145.

Reply:
Added `rm publish_log` before breaking the loop